sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StatefulProcessorHandleImpl.scala

```diff
@@ -103,22 +113,35 @@ class StatefulProcessorHandleImpl(
 
   private var currState: StatefulProcessorHandleState = CREATED
 
-  private def verify(condition: => Boolean, msg: String): Unit = {
-    if (!condition) {
-      throw new IllegalStateException(msg)
+  private val ttlExpirationMs =
+    if (ttlMode == TTLMode.ProcessingTimeTTL()) {
+      batchTimestampMs.get
+    } else if (ttlMode == TTLMode.EventTimeTTL()) {
+      eventTimeWatermarkMs.get
+    } else {
+    -1
```

Comment:
nit: indent seems iff again ?

Btw - should we just throw an exception here ?

Reply:
We do have ttlMode validation in TransformWithStateExec, so -1 value should never be used. 

We can throw an exeception, but then we need to make this a function I think. Right now, ttlExpiraionMs is created once and used across all state variables. We cannot throw the exception while initializing the class variable.